lib/webauthn/authenticator_data/attested_credential_data.rb

```diff
@@ -70,6 +77,10 @@ def public_key_position
         id_position + id_length
       end
 
+      def public_key_length
+        @public_key_length ||= CBOR.encode(Cborb.decode(data_at(public_key_position), concatenated: true).first).length
```

Comment:
Unfortunately, the only way found to split concatenated CBOR when not knowing the length was with `cborb`. As far I know `cbor` nor `libcbor` gem doesn't support that.

Reply:
Still need to give a bit more thought if there's an alternative way to achieve this without depending on another cbor-gem...

Reply:
Both [cbor](https://github.com/cabo/cbor-ruby/blob/0b6d3df203d143233ea85e46e385afca44fd1d08/doclib/cbor/unpacker.rb) and [libcbor](https://github.com/PJK/libcbor-ruby/blob/f605c1ace82b2d300537492ee905009473a6881d/lib/libcbor/streaming/buffered_decoder.rb) claim to support decoding streams which by looks of RFC 7049 encompasses concatenation: 

> In a streaming application, a data stream may be composed of a sequence of CBOR data items concatenated back-to-back

Reply:
Still couldn't figure out how to make it work with those two. Need to take a deeper look though.
If you do realize how to do it let me know :-)

Reply:
Messed around with it for a bit and figured it out 🎉 
```suggestion
        @public_key_length ||= CBOR.encode(CBOR::Unpacker.new(StringIO.new(data_at(public_key_position))).each.first).length
```

Reply:
Excellent!! :clap: :clap:

Thank you!!